### PR TITLE
gg 13964

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java [hunk 1]
```diff
@@ -52,8 +60,37 @@
...
+
+        result = Logger.getLogger(getClass().getName());
+        result.setLevel(Level.INFO);
+
+        // Adding logging to file.
+        try {
+            String namePattern = "control-utility-%g.log";
+            String absPathPattern;
+            absPathPattern = new File(JavaLoggerFileHandler.logDirectory(U.defaultWorkDirectory()), namePattern).getAbsolutePath();
+            FileHandler fileHandler = new FileHandler(absPathPattern, 1024*1024*1024, 5);
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java]: add additional space here, check once more code style guide.

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java [hunk 2]
```diff
@@ -181,27 +223,32 @@ public int execute(List<String> rawArgs) {
                 }
             }
 
+            logger.info("Command [" + commandName + "] finished with code " + EXIT_CODE_OK);
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java]: i suggest to rewrite: with code :" -- i.e. append additional ":"

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandLogger.java [hunk 3]
```diff
@@ -93,12 +94,32 @@ public static String grouped(Object... params) {
...
+     * @param e Exctption
+     * @return error message
+     */
+    public static String errorMessage(Throwable e) {
+        String msg = e.getMessage();
+
+        if (F.isEmpty(msg))
+            msg = e.getClass().getName();
+
+        if (msg.startsWith("Failed to handle request")) {
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandLogger.java]: may be: else if here?

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 4]
```diff
@@ -2432,20 +2439,21 @@ public void testCacheDistribution() throws Exception {
...
+        int firstIndex = outLines.indexOf("[next group: id=1544803905, name=default]");
+        int lastIndex  = outLines.lastIndexOf("[next group: id=1544803905, name=default]");
 
-        assertTrue(lastRowIndex > 0);
+        String dataLine = outLines.get(firstIndex + 1);
+        String userArrtDataLine = outLines.get(lastIndex + 1);
 
-        String lastRow = out.substring(lastRowIndex);
+        long commaNum = dataLine.chars().filter(i -> i==',').count();
+        long userArrtCommaNum = userArrtDataLine.chars().filter(i -> i==',').count();
```
Comment (zstan) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: need additional spaces here.

**Comment by dspavlov** [comment]
The ticket is resolved, closing PR.  
Feel free to reopen this PR if it is anyway actual.

**Pull request closed without merging**